test/unit/rand/src/RandTest.cpp

```diff
@@ -75,6 +67,7 @@ class RandTestKernel
 #if !BOOST_ARCH_PTX
             BOOST_VERIFY(std::isfinite(r));
 #endif
+            boost::ignore_unused(r);
```

Comment:
I do not think that this works here. `ignore_unused` was missing the `__device__` attribute. This leads to errors. This is the reason I added the compiler pragmas. Let's see if this works now.

Reply:
I already thought about adding my own `ignore_unused` into alpaka. However with c++ 17 we get `maybe_unused` as variable attribute which is by far superior.

Reply:
I also though it's just designed for params & templates, but it works for me, tested on gcc.

Reply:
ah year, we need a cuda-instrumented one for host - device qualifiers.

Reply:
CI proved me correct:
```
alpaka/test/unit/rand/src/RandTest.cpp:71:20: error: reference to __host__ function 'ignore_unused<float>' in __host__ __device__ function
            boost::ignore_unused(r);
                   ^
```
You should simply revert those changes and keep the pragmas.

Reply:
added `alpaka::ignore_unused` in #582 :)